mava/evaluator.py

```diff
@@ -12,329 +12,140 @@
-        env_states, timesteps = jax.vmap(env.reset)(jnp.stack(env_keys))
-        # Split keys for each core.
-        key, *step_keys = jax.random.split(key, eval_batch + 1)
-        # Add dimension to pmap over.
-        step_keys = jnp.stack(step_keys).reshape(eval_batch, -1)
-
-        # Initialise hidden state.
-        init_hstate = scanned_rnn.initialize_carry(
-            (eval_batch, config.system.num_agents),
-            config.network.hidden_state_dim,
-        )
-
-        # Initialise dones.
-        dones = jnp.zeros(
-            (
-                eval_batch,
-                config.system.num_agents,
-            ),
-            dtype=bool,
-        )
-
-        # Adding an extra batch dim for the vmapped eval functions.
-        init_hstate = init_hstate[:, jnp.newaxis, ...]
-
-        eval_state = RNNEvalState(
-            key=step_keys,
-            env_state=env_states,
-            timestep=timesteps,
-            dones=dones,
-            hstate=init_hstate,
-            step_count=jnp.zeros((eval_batch, 1)),
-            episode_return=jnp.zeros_like(timesteps.reward),
-        )
-
-        eval_metrics = jax.vmap(
-            eval_one_episode,
-            in_axes=(None, 0),
-            axis_name="eval_batch",
-        )(trained_params, eval_state)
-
-        return ExperimentOutput(
-            learner_state=eval_state,
-            episode_metrics=eval_metrics,
-            train_metrics={},
-        )
-
-    return evaluator_fn
-
-
-def make_eval_fns(
-    eval_env: Environment,
-    network_apply_fn: Union[ActorApply, RecActorApply],
-    config: DictConfig,
-    use_recurrent_net: bool = False,
-    scanned_rnn: Optional[nn.Module] = None,
-) -> Tuple[EvalFn, EvalFn]:
-    """Initialize evaluator functions for reinforcement learning.
+    """Creates a function that can be used to evaluate agents on a given environment.
 
     Args:
-        eval_env (Environment): The environment used for evaluation.
-        network_apply_fn (Union[ActorApply,RecActorApply]): Creates a policy to sample.
-        config (DictConfig): The configuration settings for the evaluation.
-        use_recurrent_net (bool, optional): Whether to use a rnn. Defaults to False.
-        scanned_rnn (Optional[nn.Module], optional): The rnn module.
-            Required if `use_recurrent_net` is True. Defaults to None.
-
-    Returns:
-        Tuple[EvalFn, EvalFn]: A tuple of two evaluation functions:
-        one for use during training and one for absolute metrics.
-
-    Raises:
-        AssertionError: If `use_recurrent_net` is True but `scanned_rnn` is not provided.
+        env: an environment that conforms to the mava environment spec.
+        act_fn: a function that takes in params, timestep, key and optionally a state
+                and returns actions and optionally a state (see `EvalActFn`).
+        config: the systems config.
+        eval_episodes: how many episodes to evaluate the agents. At least this many episodes
```

Comment:
Maybe sending the config only instead of config and eval episodes as the eval episodes lives in the config

Reply:
Is the motivation for keep it like this because you want to keep same format for absolute metric?

Reply:
Yes that was the motivation, because otherwise we have an `if is_absolute_metric` and I thought this was a bit cleaner. But I think I'll refactor this a bit so that we pass that value into the `eval_fn` instead of the `make_eval_fn`. What do you think?

Reply:
or maybe make a flag to check if it is absolute metric so you use num_absolute_metric_eval_episodes instead of eval_episodes